src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -338,35 +338,25 @@ public override async Task ValidateIntegrityAsync(SignatureContent signatureCont
 #endif
         }
 
-        public override string GetContentHashForSignedPackage(CancellationToken token)
+        public override string GetContentHash(CancellationToken token, string fallbackHashFilePath = null)
```

Comment:
Where is this called? 
If it's only in the packageextractor, then you can it to fallbackHash directly, rather than taking a disk hit.
You can also consider using a Func.

Reply:
I know this is not a new API you are adding, but you are merely unifying them, but I think it should be changed (be it in this or a different PR)

Reply:
We could avoid the disk hit by assuming that if a path is given it exists and not check for that. In our scenarios that should always be true. What do you think?